apps/obsidian/src/components/canvas/DiscourseNodePanel.tsx

```diff
@@ -0,0 +1,276 @@
+
+  useQuickReactor(
+    "dg-panel-drag-image-style",
+    () => {
+      const current = dragState.get();
+      const imageRef = rDraggingImage.current;
+      const panelRef = rPanelContainer.current;
+      if (!imageRef || !panelRef) return;
+
+      switch (current.name) {
+        case "idle":
+        case "pointing_item": {
+          imageRef.style.display = "none";
+          break;
+        }
+        case "dragging": {
+          const panelRect = panelRef.getBoundingClientRect();
+          const box = new Box(
+            panelRect.x,
+            panelRect.y,
+            panelRect.width,
+            panelRect.height,
+          );
+          const viewportScreenBounds = editor.getViewportScreenBounds();
+          const isInside = Box.ContainsPoint(box, current.currentPosition);
+          if (isInside) {
+            imageRef.style.display = "none";
+          } else {
+            imageRef.style.display = "block";
+            imageRef.style.position = "fixed";
+            imageRef.style.pointerEvents = "none";
+            imageRef.style.left = "0px";
+            imageRef.style.top = "0px";
+            imageRef.style.transform = `translate(${current.currentPosition.x - viewportScreenBounds.x - 25}px, ${current.currentPosition.y - viewportScreenBounds.y - 25}px)`;
+            imageRef.style.width = "50px";
+            imageRef.style.height = "50px";
+            imageRef.style.display = "flex";
+            imageRef.style.alignItems = "center";
+          }
+        }
+      }
+    },
+    [dragState, editor],
+  );
+
+  const nodeTypes = plugin.settings.nodeTypes;
+
+  useEffect(() => {
+    if (!focusedNodeTypeId) return;
+    const exists = !!getNodeTypeById(plugin, focusedNodeTypeId);
+    if (!exists) setFocusedNodeTypeId(null);
+  }, [focusedNodeTypeId, plugin, nodeTypes]);
+
+  const focusedNodeType = focusedNodeTypeId
+    ? getNodeTypeById(plugin, focusedNodeTypeId)
+    : null;
+
+  const displayNodeTypes = focusedNodeType ? [focusedNodeType] : nodeTypes;
+
+  const handleItemClick = (id: string) => {
+    if (didDragRef.current) return;
+    if (focusedNodeTypeId) {
+      setFocusedNodeTypeId(null);
+      return;
+    }
+    setFocusedNodeTypeId(id);
+  };
+
+  return (
+    <div className="tlui-layout__top__right">
+      <div className="tlui-style-panel__wrapper p-2" ref={rPanelContainer}>
+        <div className="flex flex-col">
+          {displayNodeTypes.map((nodeType) => (
+            <NodeTypeButton
+              key={nodeType.id}
+              nodeType={nodeType}
+              handlers={handlers}
+              didDragRef={didDragRef}
+              onClickNoDrag={() => handleItemClick(nodeType.id)}
+            />
+          ))}
+        </div>
+        <div ref={rDraggingImage}>
+          {state.name === "dragging"
+            ? (getNodeTypeById(plugin, state.nodeTypeId)?.name ?? "")
+            : null}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const NodeTypeButton = ({
```

Comment:
The hover bg is off, it is too narrow

![image.png](https://app.graphite.dev/user-attachments/assets/cb7c037e-80c3-41c9-b510-be5428d34352.png)

It should look like this

![image.png](https://app.graphite.dev/user-attachments/assets/70d1bac1-8f8d-4268-9f7e-f9112a4356c6.png)